lib/src/expression/parser.dart

```diff
@@ -0,0 +1,128 @@
+import 'package:analyzer/analyzer.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/standard_ast_factory.dart';
+import 'package:analyzer/error/listener.dart';
+import 'package:analyzer/file_system/memory_file_system.dart';
+import 'package:analyzer/src/dart/ast/token.dart';
+import 'package:analyzer/src/dart/scanner/reader.dart';
+import 'package:analyzer/src/dart/scanner/scanner.dart';
+import 'package:analyzer/src/generated/parser.dart';
+import 'package:analyzer/src/generated/source.dart';
+
+final _resourceProvider = new MemoryResourceProvider();
+
+class _ThrowingListener implements AnalysisErrorListener {
+  const _ThrowingListener();
+
+  @override
+  void onError(AnalysisError error) {
+    throw new FormatException(
+      error.toString(),
+      error.source.contents.data,
+      error.offset,
+    );
+  }
+}
+
+/// Parses a template [expression].
+Expression parseExpression(
+  String expression, {
+  bool deSugarPipes: true,
+  String sourceUrl: '/~test.dart',
+}) {
+  final source = _resourceProvider
+      .newFile(
+        sourceUrl,
+        expression,
+      )
+      .createSource();
+  final reader = new CharSequenceReader(expression);
+  final listener = const _ThrowingListener();
```

Comment:
Acknowledged. Will fix this in follow-up:

https://github.com/dart-lang/angular_ast/issues/14